src/inc/internal/MappingFileParser.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "MSIXWindows.hpp"
+#include "MsixErrors.hpp"
+#include "Exceptions.hpp"
+
+#include <map>
+#include <vector>
+
+namespace MSIX {
+
+    typedef enum : UINT32
+    {
+        Continue = 0,
+        Stop,
+        SkipSection,
+        Fail
+    } HandlerState;
+
+    enum SectionID
+    {
+        UnknownSection = -1,
+        FilesSection = 0,
+        ResourceMetadataSection = 1,
+        KeysSection = 2,
+        ExternalPackagesSection = 3,
+    };
+
+    const std::map<std::string, SectionID> KnownSectionNamesMap = 
+        {{"Files", FilesSection}, 
+        {"ResourceMetadata", ResourceMetadataSection},
+        {"Keys", KeysSection},
+        {"ExternalPackages", ExternalPackagesSection}};
+
+    class MappingFileParser final
+    {
+    public:
+        MappingFileParser();
+        void ParseMappingFile(std::string mappingFile);
+        bool IsSectionFound(SectionID sectionId);
+        std::map<std::string, std::string> GetFileList() { return this->list; }
+        std::map<std::string, std::string> GetExternalPackagesList() { return this->externalPackagesList; }

```

Comment:
As a general practice, add const at the end of the the method declaration on this type of methods. 
eg. std::map<std::string, std::string> GetFileList() const { return this->list; }

Reply:
you can also return a const reference and make it more efficient. This is only true if the caller wont modify the maps
const std::map<std::string, std::string>& GetFileList() const { return this->list; }

---
In reply to: [532771453](https://github.com/microsoft/msix-packaging/pull/395#discussion_r532771453) [](ancestors = 532771453)

Reply:
Updated to reflect this.